src/hooks/Guilds/guild/useFilteredProposalActions.tsx

```diff
@@ -0,0 +1,87 @@
+import { useMemo } from 'react';
+import useProposalCalls from 'hooks/Guilds/guild/useProposalCalls';
+import {
+  DecodedAction,
+  SupportedAction,
+} from 'old-components/Guilds/ActionsBuilder/types';
+
+const isApprovalCall = (action: DecodedAction) => {
+  // TODO: improve this with better conditions. At read point decodedAction has no other things to validate than generic call && approval event in contract interface

```

Comment:
We definitely should introduce a new SupportedAction type for approvals and flag it while decoding. Can do it in a different PR if it's too much work.

Reply:
I'll make an issue for this, it does I think make sense to flag it as early as possible. We can also as Milton suggests in the comment make it more reliable to identify the spending calls